### PR TITLE
Kivy baseclass inbox

### PR DESCRIPTION
**Pull request by shekhar-cis**
Add kivy inbox screen

### CODE REVIEW

File: src/bitmessagekivy/baseclass/inbox.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        """This method set's default address"""
+        if self.kivy_state.association == "":
+            if self.kivy_running_app.identity_list:
+                self.kivy_state.association = self.kivy_running_app.identity_list[0]
+
+    def init_ui(self, dt=0):
+        """Clock schdule for method inbox accounts"""
+
+    def refresh_callback(self, *args):
+        """Method updates the state of application,
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/inbox.py:46]: incomplete docstring

File: src/bitmessagekivy/baseclass/inbox.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        While the spinner remains on the screen"""
+
+        def refresh_callback(interval):
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/inbox.py:49]: rename function

File: src/bitmessagekivy/baseclass/inbox.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            """Method used for loading the inbox screen data"""
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/inbox.py:50]: better docstring

File: src/bitmessagekivy/baseclass/inbox.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        Clock.schedule_once(self.init_ui, 0)
+
+    def set_defaultAddress(self):
+        """This method set's default address"""
+        if self.kivy_state.association == "":
+            if self.kivy_running_app.identity_list:
+                self.kivy_state.association = self.kivy_running_app.identity_list[0]
+
+    def init_ui(self, dt=0):
+        """Clock schdule for method inbox accounts"""
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/inbox.py:43]: typo

File: src/bitmessagekivy/baseclass/inbox.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    """Inbox Screen class for kivy Ui"""
+
+    queryreturn = ListProperty()
+    has_refreshed = True
+    account = StringProperty()
+    no_search_res_found = "No search result found"
+    label_str = "Yet no message for this account!"
+
+    def __init__(self, *args, **kwargs):
+        """Method Parsing the address"""
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/inbox.py:29]: docstring

File: src/bitmessagekivy/baseclass/inbox.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        super(Inbox, self).__init__(*args, **kwargs)
+        self.kivy_running_app = App.get_running_app()
+        self.kivy_state = kivy_state_variables()
+        self.image_dir = load_image_path()
+        Clock.schedule_once(self.init_ui, 0)
+
+    def set_defaultAddress(self):
+        """This method set's default address"""
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/inbox.py:37]: update docstring

**Review by PeterSurda — CHANGES_REQUESTED** [review]

**Review by PeterSurda — CHANGES_REQUESTED** [review]
Fix element paths

File: src/bitmessagekivy/tests/test_create_random_address.py [hunk 7]
```diff
@@ -23,38 +21,42 @@ def test_landing_screen(self):
...
+        )
         self.assertExists("//ScreenManager[@current=\"random\"]", timeout=5)
 
     @ordered
     def test_generate_random_address_label(self):
         """Creating New Adress For New User."""
         # Checking the Button is rendered
         self.assertExists(
-            '//RandomBoxlayout/BoxLayout[0]/AnchorLayout[1]/MDTextField[@hint_text=\"Label\"]', timeout=2)
+            '//Random//RandomBoxlayout//BoxLayout[0]/AnchorLayout[1]/MDTextField[@hint_text=\"Label\"]', timeout=1
```
Comment (PeterSurda) [inline src/bitmessagekivy/tests/test_create_random_address.py:40]: maybe remove `/BoxLayout[0]/AnchorLayout[1]`

File: src/bitmessagekivy/tests/test_create_random_address.py [hunk 8]
```diff
@@ -23,38 +21,42 @@ def test_landing_screen(self):
...
+        )
         # Click on Label Text Field to give address Label
         self.cli.wait_click(
-            '//RandomBoxlayout/BoxLayout[0]/AnchorLayout[1]/MDTextField[@hint_text=\"Label\"]', timeout=2)
+            '//Random//RandomBoxlayout/BoxLayout[0]/AnchorLayout[1]/MDTextField[@hint_text=\"Label\"]', timeout=2)
```
Comment (PeterSurda) [inline src/bitmessagekivy/tests/test_create_random_address.py:44]: also

File: src/bitmessagekivy/tests/test_create_random_address.py [hunk 9]
```diff
@@ -23,38 +21,42 @@ def test_landing_screen(self):
...
-        self.assertExists('//RandomBoxlayout//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=3)
+        self.assertExists('//Random//RandomBoxlayout//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=3)
         # Click on Proceed Next button to generate random Address
-        self.cli.wait_click('//RandomBoxlayout//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=3)
+        self.cli.wait_click('//Random//RandomBoxlayout//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=3)
         # Checking "My Address" Screen after creating a address
         self.assertExists("//ScreenManager[@current=\"myaddress\"]", timeout=5)
         # Checking the new address is created
-        self.assertExists('//MDList[0]/CustomTwoLineAvatarIconListItem', timeout=10)
+        self.assertExists('//MyAddress//MDList[0]/CustomTwoLineAvatarIconListItem', timeout=10)
```
Comment (PeterSurda) [inline src/bitmessagekivy/tests/test_create_random_address.py:58]: the `MDList` should have and identifier, not an index

Comment (PeterSurda) [reply src/bitmessagekivy/tests/test_create_random_address.py:58]: Something like:
`//MyAddress//MDList[@id="ml"]/CustomTwoLineAvatarIconListItem[0]`

File: src/bitmessagekivy/tests/test_create_random_address.py [hunk 10]
```diff
@@ -63,13 +65,11 @@ def test_set_default_address(self):
         # This is for opening side navbar
         self.open_side_navbar()
         # Click to open Address Dropdown
-        self.cli.wait_click('//NavigationItem[0]/CustomSpinner[0]', timeout=5)
+        self.cli.wait_click('//NavigationItem[0]/IdentitySpinner[0]', timeout=5)
```
Comment (PeterSurda) [inline src/bitmessagekivy/tests/test_create_random_address.py:68]: - better `NavigationItem` as there are `NavigationItem`s at two different levels
- reference `IdentitySpinner` by key, not by index, e.g. `IdentitySpinner[@id=\"identity_dropdown\"]`

File: src/bitmessagekivy/tests/test_create_random_address.py [hunk 11]
```diff
@@ -63,13 +65,11 @@ def test_set_default_address(self):
...
         # Checking the dropdown option is exist
         self.assertExists('//MySpinnerOption[0]', timeout=5)
-        is_open = self.cli.getattr('//NavigationItem[0]/CustomSpinner[@is_open]', 'is_open')
+        is_open = self.cli.getattr('//NavigationItem[0]/IdentitySpinner[@is_open]', 'is_open')
```
Comment (PeterSurda) [inline src/bitmessagekivy/tests/test_create_random_address.py:71]: here also at the very least add a key to `IdentitySpinner`

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/bitmessagekivy/tests/test_create_random_address.py (hunk 9 above)
Comment (shekhar-cis) [reply src/bitmessagekivy/tests/test_create_random_address.py:59]: I debugged MDList, yes we can add like `name:mdlist_name` but can not get with `self.cli.getattr()`.
I tried this approach with `id` and `name` but it is not working.
`//MyAddress//MDList[@name="mdlist_name"]/CustomTwoLineAvatarIconListItem[0]`

File: src/bitmessagekivy/tests/test_create_random_address.py [hunk 12]
```diff
@@ -63,13 +66,11 @@ def test_set_default_address(self):
         # This is for opening side navbar
         self.open_side_navbar()
         # Click to open Address Dropdown
-        self.cli.wait_click('//NavigationItem[0]/CustomSpinner[0]', timeout=5)
+        self.cli.wait_click('//IdentitySpinner[0]', timeout=1)
```
Comment (shekhar-cis) [inline src/bitmessagekivy/tests/test_create_random_address.py:69]: Here also, i added `name:` in  `IdentitySpinner` but getting assertion error and can not get `name` and `id`

File: src/bitmessagekivy/main.kv [hunk 13]
```diff
@@ -104,10 +104,13 @@
                     text: app.tr._('Accounts')
                     height:"35dp"
                 NavigationItem:
+                    text: 'dropdown_nav_item'
                     height: dp(48)
                     IdentitySpinner:
                         id: identity_dropdown
                         pos_hint:{"x":0,"y":0}
+                        name: "identity_dropdown"
```
Comment (PeterSurda) [inline src/bitmessagekivy/main.kv:112]: why is it duplicate?

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/bitmessagekivy/base_navigation.py [hunk 14]
```diff
@@ -108,3 +108,4 @@ def __init__(self, *args, **kwargs):
         self.dropdown_cls.max_height = Window.size[1] / 3
         self.values = list(addr for addr in config.addresses()
                            if config.getboolean(str(addr), 'enabled'))
+        self.name = 'identity_dropdown_name'
```
[the remark this answers is not in the document]
Comment (shekhar-cis) [reply src/bitmessagekivy/base_navigation.py]: Removed

File: src/bitmessagekivy/main.kv (hunk 13 above)
Comment (shekhar-cis) [reply src/bitmessagekivy/main.kv:112]: Fixed

File: src/bitmessagekivy/baseclass/maildetail.py [hunk 15]
```diff
@@ -154,7 +154,7 @@ def delete_mail(self):
             self.parent.screens[2].ids.ml.clear_widgets()
             self.parent.screens[2].loadSent(self.kivy_state.selected_address)
         elif self.kivy_state.detail_page_type == 'inbox':
```
Comment (PeterSurda) [inline src/bitmessagekivy/baseclass/maildetail.py:156]: for variables with a fixed number of valid values, it's better to use `enum`

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**